src/Ideal.jl

```diff
@@ -29,19 +140,20 @@ function *(x::RingElement, R::Ring)
   return ideal(R, x)
 end
 
-function ideal(x::NCRingElement; kw...)
-  return ideal(parent(x), x; kw...)
+function *(I::Ideal{T}, p::T) where T <: RingElement
+  iszero(p) && return similar(I, T[])
+  return similar(I, [v*p for v in gens(I)])
```

Comment:
so here, if we assume `Ideal` means "two sided", and also "there is one (two-sided) ideal type per ring", and also `ideal(R, xs)` produces that ideal type, then we can write here
```suggestion
  return ideal(base_ring(I), [v*p for v in gens(I)])
```
and indeed that's what I did before the second commit in this PR, and I'd be happy to back to it if we can agree on this scheme.